src/euler/staking/StakeableEulerERC4626.sol

```diff
@@ -64,8 +64,8 @@ contract StakeableEulerERC4626 is EulerERC4626, Owned {
     }
 
     /// @notice Allows owner to set or update a new staking contract. Will claim rewards from previous staking if available
-    function updateStakingAddress(uint256 rewardIndex, address /*recipient*/) external onlyOwner {
-        // TODO: Claim rewards if was staking on another contract
+    function updateStakingAddress(uint256 rewardIndex, address recipient) external onlyOwner {
+        _stopStaking(recipient);
 
         IRewardsDistribution.DistributionData memory data = rewardsDistribution.distributions(rewardIndex);
         IStakingRewards stakingRewards_ = IStakingRewards(data.destination);
```

Comment:
Should we set approve 1 to the previous one? Just if there is some safety concerns?

Reply:
Good idea. I'm setting the approval to 0 for the previous one